pkg/controller/namespace/namespace_controller.go

```diff
@@ -55,16 +57,36 @@ func NewNamespaceController(kubeClient client.Interface, resyncPeriod time.Durat
 		framework.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				namespace := obj.(*api.Namespace)
-				err := syncNamespace(kubeClient, *namespace)
-				if err != nil {
-					glog.Error(err)
+				if err := syncNamespace(kubeClient, *namespace); err != nil {
+					if estimate, ok := err.(*contentRemainingError); ok {
+						go func() {
+							t := estimate.Estimate/2 + 1
```

Comment:
The estimate is the maximum, most things will terminate fast and a few will
go longer.  I *believe* the distribution will be binomial with a strong
peak around a few seconds, but our default is 30 seconds for pods.   So I
didn't want to wait 30, and /2 felt like a good compromise.  +1 was to
ensure that we wouldn't wait 0 in scale up cases.  Will add a comment.

On Mon, Aug 10, 2015 at 3:21 AM, Nikhil Jindal <notifications@github.com>
wrote:

> In pkg/controller/namespace/namespace_controller.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12406#discussion_r36606759>
> :
>
> > @@ -55,16 +57,36 @@ func NewNamespaceController(kubeClient client.Interface, resyncPeriod time.Durat
> >  		framework.ResourceEventHandlerFuncs{
> >  			AddFunc: func(obj interface{}) {
> >  				namespace := obj.(*api.Namespace)
> > -				err := syncNamespace(kubeClient, *namespace)
> > -				if err != nil {
> > -					glog.Error(err)
> > +				if err := syncNamespace(kubeClient, *namespace); err != nil {
> > +					if estimate, ok := err.(*contentRemainingError); ok {
> > +						go func() {
> > +							t := estimate.Estimate/2 + 1
>
> If the remaining content is going to take 10 secs to delete, then why not
> wait 10 secs? Why wait just 6 secs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12406/files#r36606759>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift